### ISSUE TITLE
Inconsistent formatting for $this vs other variable for fluent interfaces

### ISSUE THREAD

**Issue by karptonite** [prettier/plugin-php#1386]
**Prettier 2.0.4**

**PHP Plugin 0.14.0**

```sh
# Options (if any):
(all default)
```

**Input:**

```php
$that->shouldReceive( 'signUp' )->with( anInstanceOf( 'Foo\Bar\Baz' ), $this->app['foo.bar.baz']->getEmail())->once();

$this->shouldReceive( 'signUp' )->with( anInstanceOf( 'Foo\Bar\Baz' ), $this->app['foo.bar.baz']->getEmail())->once();
```

**Output:**

```php
$that
    ->shouldReceive('signUp')
    ->with(anInstanceOf('Foo\Bar\Baz'), $this->app['foo.bar.baz']->getEmail())
    ->once();

$this->shouldReceive('signUp')
    ->with(anInstanceOf('Foo\Bar\Baz'), $this->app['foo.bar.baz']->getEmail())
    ->once();
```

**Expected behavior:**
```php
$that->shouldReceive('signUp')
    ->with(anInstanceOf('Foo\Bar\Baz'), $this->app['foo.bar.baz']->getEmail())
    ->once();

$this->shouldReceive('signUp')
    ->with(anInstanceOf('Foo\Bar\Baz'), $this->app['foo.bar.baz']->getEmail())
    ->once();
```

Namely, `$that` shouldn't be on a line by itself in the first example. It should be formatted the same way as with `$this`.
Example [here](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCACsAOj4gWMCG3iyVAYXQBbH2wAFAAkA-BdBQxAOVwBjKABlGABPNWswbAgAWgAmJxBoAEdBeGgA12V-PTkFEGIRCBgytgq2KoMAXxYoCngYdBQ0eCxJUDZsGDaRTtrMMEEJvBAAHgB+YzAAHUwtrYASGjYYTIA+bGp0QVkKACUoOPgANygACgACAHIlZUwAVTB314AShOAHdetQ3mxMABJHDsTBxZBSN7vABi6HQGw2ACE2BAsbiAF4AwFMV77ajwbAnNhgMAAbXeUgxdg4eNZbGJAF0TspYABRfzuZ6A4HHLBxEUAbh2mApVJOZwuV1u9yeKK+v3+QNB4MhMLhUMRyI+6MxOLxBM5JLJ8upx1pDKZLLZEA53N5AqFshFYolL0BMu2mBA7SAA).

**Comment by czosel** [comment]
Hi @karptonite, thanks for the issue report! I didn't know that we're having special treatment for `$this`, so did some digging and it turns out that it was introduced in https://github.com/prettier/plugin-php/pull/372 as a port from prettier for JS, which [does the same](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMAWBLAzgOgGYSFHEmmEAUAlDgEYCGDjTzLrDATlbWz2wF4CqAHSgj0dGPjLTpXerwVtO1eYoUC+VEABoQEAA4wM0LMlB127CAHcAChYSmUdAG4QMAEx0ga7OmABrOBgAZX1-DCgAc2QYdgBXOF00GABbABsAdUx4LHCwOBDHDCMXEoBPZHAsU11IrDh2GFs-KNS6ZDw6dIbdACssAA8AIT9A4JC6VLgAGUi4Tu7ekAHBkMio9LgARXiIeEWepJBw9gb2Krx0m299dkiYTM90ZAAOAAZdO4gGzL99Kp3ODnFwLXQARz28BaBicIDoWAAtFA4HAPGjvOw4JCMFiWnQ2h0kF0jroGqkMLEEscsBstrt9gtiUtjjA6DQnh4XkgAEy6OJ0DDpDYAYQgqXaVSg0DBIHiDQAKuynCTli5EgBJKDo2AhMD3QwAQW1IRg5S2h2WCJgluOHggYFtugg8Rg+ldPNtAF8vUA). 

I have the feeling that PHP and JS are quite comparable when it comes to what `this` means and how it is used, so the reasoning that made Prettier (for JS) handle it this way probably applies to the PHP plugin as well. I don't know the reason though, and would be curious to find out! 

cc @evilebottnawi

**Comment by karptonite** [comment]
Hmm... Maybe I should report this in Prettier itself, then. I can't imagine any really good reason for treating them differently.

**Comment by evilebottnawi** [comment]
Don't know :smile: https://github.com/prettier/prettier/issues/777

**Comment by karptonite** [comment]
Interestingly, prettier, (but not prettier-plugin-php) has a heuristic to not break the chain if the variable name is short enough. See: https://github.com/prettier/prettier/issues/8003#issuecomment-613553483
So, for example, prettier PHP format the following code like this:
```php
$db
    ->setQuery()
    ->setValue()
    ->execute();
```
but by the heuristic they use for the js version, the `setQuery` would be on the same line as $db. There is still discussion of this rule in general in that prettier issue, but regardless, it seems like a good idea to adopt the same heuristic comparing variable length to tab width here.

**Comment by czosel** [comment]
I totally agree! 👍 the only reason I can imagine that we’re not using the same heuristic is that is has been added after we ported the code over from Prettier.

**Comment by karptonite** [comment]
@czosel should I add a new feature request to support the heuristic from prettier mentioned above? I worked for a bit on a pull request, but it turned out to be more complicated than I expected, especially with my lack of experience in this code base.
.

**Comment by czosel** [comment]
@karptonite Sure! Feel free to also open a PR with your attempt to implement it, as it might help as a starting point :wink:

**Comment by czosel** [comment]
I’ll go ahead and close this since the heuristic from prettier has landed in 0.14.3.
